### PR TITLE
Only enter stations close

### PR DESCRIPTION
**Pull request by johan-t**
# Pull Request Template

## Description

Checks the users location and verifies that it is within 1km of the station. Else it will show an error.

Fixes: #64 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Ran on mobile and tested with station out of reach and station in reach.

## Checklist:

Before submitting your PR, please review the following checklist:

- [x] I have performed a self-review of my own code or materials.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

## Additional Information:

Any additional information that you believe is important and relevant to the review of this PR.

### CODE REVIEW

File: src/components/ReportForm/ReportForm.tsx [hunk 1]
```diff
@@ -52,6 +43,23 @@ const redHighlight = (text: string) => {
 	return <>{text}<span className='red-highlight'>*</span></>
 }
 
+function calculateDistance(lat1: number, lon1: number, lat2: number, lon2: number) {
```
Comment (brandesdavid) [inline src/components/ReportForm/ReportForm.tsx:46]: Maybe put this math function in another folder, as it doesn't depend on any states in this component and could be reused for further features

File: src/components/ReportForm/ReportForm.tsx [hunk 2]
```diff
@@ -87,6 +100,28 @@ const ReportForm: React.FC<ReportFormProps> = ({
...
+		if (!stationInput) return false;
+
+		const userLocation = await getPosition();
+		const station = stationsList[stationInput.value];
+		if (!station) return false;
+
+		const distance = userLocation ? calculateDistance(userLocation[0], userLocation[1], station.coordinates.latitude, station.coordinates.longitude): 0;
+
+		// Too far from the selected station
+		if (distance > 1) {
```
Comment (brandesdavid) [inline src/components/ReportForm/ReportForm.tsx:116]: If refactoring the function calculateDistance, maybe write into the comment that 1 is in km

**Review by brandesdavid — COMMENTED** [review]
Maybe refactor the code and then merge, however you like. looks really good

**Comment by johan-t** [comment]
@brandesdavid Thanks for the feedback will do as you have suggested

**Pull request merged**